src/main/java/com/fasterxml/jackson/core/json/UTF8JsonGenerator.java

```diff
@@ -1344,6 +1344,9 @@ private final void _writeStringSegments(String text, boolean addQuotes) throws I
 
         while (left > 0) {
             int len = Math.min(_outputMaxContiguous, left);
+            if (_isStartOfSurrogatePair(text.charAt(len-1))) {
```

Comment:
Could you add a brief comment above, on why this is done (to avoid incorrect splitting of surrogate pairs so encoding works).

Reply:
added